File: src/GameCore/Tag/TagFactory.cs
namespace GameCore.Tag;

/// <summary>
/// TagFactory — 从 TagDef 创建运行时 Tag 实例。
/// recalculateTags 第四步调用 createAll。
/// </summary>
public sealed class TagFactory
{
    private readonly TagRegistry _registry;

    public TagFactory(TagRegistry registry)
    {
        _registry = registry;
    }

    public Tag Create(string tagId) => new(_registry.Get(tagId));

    public List<Tag> CreateAll(IEnumerable<string> tagIds)
        => tagIds.Select(Create).ToList();
}

File: canglan-backend/canglan-core/src/main/java/com/canglan/core/tag/TagFactory.java
package com.canglan.core.tag;

import java.util.ArrayList;
import java.util.List;

/**
 * TagFactory — 从 TagDef 创建运行时 Tag 实例。
 * recalculateTags 第三步调用 createAll。对应 C# TagFactory。
 */
public final class TagFactory {

    private final TagRegistry registry;

    public TagFactory(TagRegistry registry) {
        this.registry = registry;
    }

    public Tag create(String tagId) {
        return new Tag(registry.get(tagId));
    }

    public List<Tag> createAll(Iterable<String> tagIds) {
        List<Tag> tags = new ArrayList<>();
        for (String id : tagIds) tags.add(create(id));
        return tags;
    }
}
